### PR TITLE
Improve sidebar look

### PR DESCRIPTION
**Pull request by viktor-yakubiv**
It styles secondary navigation at the sidebar. Drawer's interface is not really good, I hope we will improve it later. Also, I would appreciate any feedback about naming, I borrowed my one from [Material Design](https://material.io/components/navigation-drawer/).

It breaks navigation transitions because of Link removing. I will fix it in separate PR.

### CODE REVIEW

**Comment by viktor-yakubiv** [comment]
@Joozty this is ready for the review,

File: design/drawer/item.jsx [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  </Tag>
+)
+
+const TreeItem = ({ children, ...passProps }) => (
+  <li>
+    <FlatItem {...passProps}>{children}</FlatItem>
+  </li>
+)
+
+const Item = ({ children, tag = 'li', ...passProps }) => {
```
Comment (Joozty) [inline design/drawer/item.jsx]: Interesting sometimes you call it passProps and sometimes restProps? Is there any difference?

Comment (viktor-yakubiv) [reply design/drawer/item.jsx]: 'rest' if I actually destructurise, pass if I just forward without destructuring.

This particular case is special, I destrucutrise to make correct forwarding.

Comment (viktor-yakubiv) [reply design/drawer/item.jsx]: Maybe I should always call it `forwardProps`. What do you think?

File: components/application/activity-select.css [hunk 2]
```diff
@@ -1,32 +1,21 @@
-.sidebar-navigation,
-.sidebar-navigation li > .route {
-  padding: 0;
-  margin: 0;
-  text-indent: 0;
-  list-style-type: none;
+/* Should be deleted after fix in Icon component */
```
Comment (Joozty) [inline components/application/activity-select.css]: Would be nice to create task for it. :)

File: design/drawer/item.jsx (hunk 1 above)
Comment (Joozty) [reply design/drawer/item.jsx]: forwardProps sounds better but passProps is good too

**Pull request merged**